### PR TITLE
feat, bam to cram

### LINKED ISSUE #286: Option to deliver results as CRAM instead of BAM for panel runs
**Is your feature request related to a problem? Please describe.**
BAM output files take too much storage space and it is a recurring storage issue on Hasta.

**Describe the solution you'd like**
Convert BAM files to CRAM.

**Describe alternatives you've considered**
None.

**Additional context**
CRAM index files should also be readily available

**Expected output for the feature**
`tumor.merged.bam` to `tumor.merged.cram`. And all `bai` files to `crai`

**Current BALSAMIC version**
`4.1.0`

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

- Fixes #286 

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [x] Code review
- [x] New code is executed and covered by tests
- [x] Tests pass

### CODE REVIEW

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 1]
```diff
@@ -50,12 +51,13 @@ rule mergeBam_normal:
     "source activate {params.conda}; "
     "picard AddOrReplaceReadGroups {params.picard} INPUT={input.bamN} OUTPUT={output.bamN}; "
     "samtools index {output.bamN}; "
-    "samtools view -c {output.cramN} {output.bamN}; "
+    "samtools view -C -o {output.cramN} -T {input.fasta} {output.bamN} ; "
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]:
```suggestion
    "samtools view -C -o {output.cramN} -T {input.fasta} {output.bamN}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 2]
```diff
@@ -71,7 +73,7 @@ rule mergeBam_tumor:
     "source activate {params.conda}; "
     "picard AddOrReplaceReadGroups {params.picard} INPUT={input.bamT} OUTPUT={output.bamT}; "
     "samtools index {output.bamT}; "
-    "samtools view -c {output.cramT} {output.bamT}; "
+    "samtools view -C -o {output.cramT} -T {input.fasta} {output.bamT} ; "
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]:
```suggestion
    "samtools view -C -o {output.cramT} -T {input.fasta} {output.bamT}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 3]
```diff
@@ -44,14 +44,15 @@ rule mergeBam_normal:
...
+  threads: get_threads(cluster_config, "mergeBam_normal")
   singularity: singularity_image
   benchmark:
     benchmark_dir + 'mergeBam_normal_' + "{mysample}.mergebam_normal.tsv".format(mysample = normal_sample)
   shell:
     "source activate {params.conda}; "
     "picard AddOrReplaceReadGroups {params.picard} INPUT={input.bamN} OUTPUT={output.bamN}; "
     "samtools index {output.bamN}; "
-    "samtools view -C -o {output.cramN} -T {input.fasta} {output.bamN} ; "
+    "samtools view --threads {threads} -C -o {output.cramN} -T {input.fasta} {output.bamN} ; "
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule:55]:
```suggestion
    "samtools view --threads {threads} -C -o {output.cramN} -T {input.fasta} {output.bamN}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule [hunk 4]
```diff
@@ -66,14 +67,15 @@ rule mergeBam_tumor:
...
+  threads: get_threads(cluster_config, "mergeBam_tumor")
   singularity: singularity_image
   benchmark:
     benchmark_dir + 'mergeBam_tumor_' + "{mysample}.mergebam_tumor.tsv".format(mysample = tumor_sample)
   shell:
     "source activate {params.conda}; "
     "picard AddOrReplaceReadGroups {params.picard} INPUT={input.bamT} OUTPUT={output.bamT}; "
     "samtools index {output.bamT}; "
-    "samtools view -C -o {output.cramT} -T {input.fasta} {output.bamT} ; "
+    "samtools view --threads {threads} -C -o {output.cramT} -T {input.fasta} {output.bamT} ; "
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule:78]:
```suggestion
    "samtools view --threads {threads} -C -o {output.cramT} -T {input.fasta} {output.bamT}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule (hunk 1 above)
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]:
```suggestion
    "samtools view -C -o {output.cramN} -T {input.fasta} {output.bamN}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/mergetype.rule (hunk 2 above)
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/mergetype.rule]:
```suggestion
    "samtools view -C -o {output.cramT} -T {input.fasta} {output.bamT}; "
```

**Comment by hassanfa** [comment]
Ready to review again.

**Pull request merged**